core/src/main/java/org/eclipse/krazo/binding/convert/impl/DoubleConverter.java

```diff
@@ -37,7 +37,7 @@ public boolean supports(Class<Double> rawType) {
     @Override
     public ConverterResult<Double> convert(String value, Class<Double> rawType, Locale locale) {
 
-        Double defaultValue = Double.TYPE.equals(rawType) ? 0.0 : null;
+        Double defaultValue = Double.TYPE.equals(rawType) ? 0.0 : Double.NaN;
```

Comment:
I'm not sure about this one. Actually I think this violates the spec. The spec states in "3.4.1 Numeric types":

> Empty strings are either converted to null or to the default value of the corresponding primitive data type. 

The problematic part about our data binding integration is that JAX-RS ALWAYS expects a result from a converter if it doesn't throw an exception (and therefore aborts the request lifecycle with a 400 status codes). Therefore, our converter MUST return a result which JAX-RS can bind to the corresponding field/parameter even in case of validation errors and empty submissions.

Reply:
If that's what the spec says fine :-) I'll revert this back to null

I was aiming for a more consistent behavior on the edge cases. Right now we return the NaN in most of those cases (with float and double).

Reply:
Really? See: 

https://github.com/eclipse-ee4j/krazo/blob/05d4c03bf8f7e915478022727dedc3b3471365d8/core/src/main/java/org/eclipse/krazo/binding/convert/impl/DoubleConverter.java#L40

The basic idea was to return the data type "initialization default value" in case of empty strings or validation errors. As described above, JAX-RS requires us to return a value or it will abort the request. Actually the user would expect that no binding happens at all if a binding error occurs, but JAX-RS forces us to return a value from the converter. And by returning the data type default value, it almost looks like JAX-RS doesn't perform any data binding in these cases, also it actually does (but the user won't notice).

I know that this feels dirty, but I didn't find any other way to implement the binding/validation error handling as required by the spec.